zetteldeft.el

```diff
@@ -326,52 +333,89 @@ Don't forget to add `\\n' at the beginning to start a new line."
   :type 'string
   :group 'zetteldeft)
 
-(defun zetteldeft--insert-title ()
-  "Insert filename of current zd note, stripped from its ID.
+(defun zetteldeft--insert-title (title)
+  "Insert TITLE as title in file.
 Prepended by `zetteldeft-title-prefix' and appended by `zetteldeft-title-suffix'."
   (zetteldeft--check)
   (insert
     zetteldeft-title-prefix
-    (zetteldeft--lift-file-title (file-name-base (buffer-file-name)))
+    title
     zetteldeft-title-suffix))
 
 (defun zetteldeft--lift-file-title (zdFile)
   "Return the title of a zetteldeft note.
 ZDFILE should be a full path to a note."
-  (let ((baseName (file-name-base zdFile)))
-    (replace-regexp-in-string
-     (concat zetteldeft-id-regex "[[:space:]]")
-     "" baseName)))
-
-(defun zetteldeft-file-rename ()
-  "Rename the current file via the deft function.
-Use this on files in the `deft-directory'."
-  (interactive)
+  (let ((deft-use-filename-as-title nil)
+	contents title)
+    (with-current-buffer (get-buffer-create "*Zetteldeft temp*")
+      (insert-file-contents zdFile nil nil nil t)
+      (setq contents (concat (buffer-string))))
+    (setq title (deft-parse-title zdFile contents))
+    (kill-buffer "*Zetteldeft temp*")
+    title))
+
+(defun zetteldeft--rename (cusprompt fn-gen update-title)
```

Comment:
This function name should indicate that it both (1) renames the file and (2) updates the title in the buffer.

Also, passing prompt/action functions as arguments to this function is a flexible approach, but merits more commentary to indicate how it's designed and used. I had to read the code more than once to see what was going on here.

Reply:
I've updated the function name, and provided a more detailed explanation in the doc string.